client/my-sites/marketing/tools/google-adwords.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import page from 'page';
+import PropTypes from 'prop-types';
+import React, { Fragment, FunctionComponent } from 'react';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { addQueryArgs } from 'lib/url';
+import Button from 'components/button';
+import { getSelectedSite } from 'state/ui/selectors';
+import GoogleVoucherDetails from 'my-sites/checkout/checkout-thank-you/google-voucher';
+import { isPremium, isBusiness, isEcommerce, isEnterprise } from 'lib/products-values';
+import isSiteAtomic from 'state/selectors/is-site-automated-transfer';
+import MarketingToolsFeature from './feature';
+import QuerySiteVouchers from 'components/data/query-site-vouchers';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface ConnectProps {
+	isAtomic: boolean;
+	isJetpack: boolean;
+	isPremiumOrHigher: boolean;
+	siteId: number | null;
+	siteSlug: string | null;
+}
+
+export const GoogleAdwordsCard: FunctionComponent< ConnectProps > = ( {
+	isAtomic,
+	isJetpack,
+	isPremiumOrHigher,
+	siteId,
+	siteSlug,
+} ) => {
+	const translate = useTranslate();
+	if ( ( isJetpack && ! isAtomic ) || ! siteId || ! siteSlug ) {
+		return null;
+	}
+
+	const handleUpgradeClick = () => {
+		page( addQueryArgs( { plan: 'PLAN_PREMIUM' }, `/plans/${ siteSlug }` ) );
+	};
+
+	const renderButton = () => {
+		if ( isPremiumOrHigher ) {
+			return <GoogleVoucherDetails />;
+		}
+		return <Button onClick={ handleUpgradeClick }>{ translate( 'Upgrade To Premium' ) }</Button>;
+	};
+
+	return (
+		<Fragment>
+			<QuerySiteVouchers siteId={ siteId } />
+			<MarketingToolsFeature
+				title={ translate( 'Advertise with your $100 Google Adwords credit' ) }
+				description={ translate(
+					'Advertise your site where most people are searching: Google. You’ve got a $100 credit with Google Adwords to drive traffic to your most important pages.'
+				) }
+				imagePath="/calypso/images/illustrations/marketing.svg"
+			>
+				{ renderButton() }
+			</MarketingToolsFeature>
+		</Fragment>
+	);
+};
+
+GoogleAdwordsCard.propTypes = {
+	isAtomic: PropTypes.bool.isRequired,
+	isJetpack: PropTypes.bool.isRequired,
+	isPremiumOrHigher: PropTypes.bool.isRequired,
+	siteId: PropTypes.number,
+	siteSlug: PropTypes.string,
+};
```

Comment:
We discussed this in Slack and decided that unless we know for sure that this will never be used outside of a Typescript file we still need it.  @dmsnell Can you give us a ruling here?

Reply:
what uses do we have for it?

I think some of our documentation tries to infer it from PropTypes but there are definite problems with that and it'd be better to fix our tooling to use the types themselves rather than `PropTypes`

1. There are _no_ constraints on `PropTypes`. They _will_ and _do_ get out of sync with the code. For the same reason we should probably rip out JSDoc comments whose purpose is showing types. TypeScript enforces the constraints so `interface Props` is as true as the actual code.
2. Most of our `PropTypes` were incomplete to begin with. There is an embarrassing deficiency in the vast majority of our `PropTypes` declarations. It's only embarrassing from the point of view of communicating "this component needs these things." In truth the ones that actually get listed are there because there was some confusion or anticipation of confusion about a specific subset of properties.
3. `PropTypes` are a runtime type guard where TypeScript type annotations are stripped from the build. We no longer need to burden our customers with the deficiency of our language.
4. Although `PropTypes` _can_ describe more expressive types by creating them as exported objects the system gets somewhat unwieldy and IDEs don't support them well. Better just to let the strong tool provide the information than introduce two copies of what should be the same data but never is.

I wasn't privy to the Slack conversation but I'd like to hear your concerns.

Reply:
So lets delete.